lib/representer.ex

```diff
@@ -277,6 +277,47 @@ defmodule Representer do
     do_use_existing_placeholders(node, represented)
   end
 
+  # module attributes that hold module or function names inside of key-value pairs.
+  # Note that module attributes that hold module or function names outside of key-value pairs are excluded from this list.
+  @attributes_with_key_value_pairs_that_hold_module_or_function_names ~w(compile optional_callbacks dialyzer)a
+  defp do_use_existing_placeholders({:@, meta, [{name, meta2, value}]}, represented)
+       when name in @attributes_with_key_value_pairs_that_hold_module_or_function_names do
+    handle_list = fn list, represented ->
+      {new_list, new_represented} =
+        Enum.reduce(list, {[], represented}, fn elem, {acc, represented} ->
+          {elem, represented} =
+            case elem do
+              {_, _} -> use_existing_placeholders_on_key_value_pair(elem, represented)
+              _ -> {elem, represented}
+            end
+
+          {[elem | acc], represented}
+        end)
+
+      {Enum.reverse(new_list), new_represented}
+    end
+
+    {value, represented} =
+      case value do
+        # e.g. @dialyzer {:nowarn_function, function_name: 0}
+        [{elem1, elem2}] when is_list(elem2) ->
+          {elem2, represented} = handle_list.(elem2, represented)
+          {[{elem1, elem2}], represented}
+
+        # e.g. @optional_callbacks [function_name1: 0]
+        [elem1] when is_list(elem1) ->
+          {elem1, represented} = handle_list.(elem1, represented)
+          {[elem1], represented}
+
+        value ->
+          {value, represented}
+      end
+
+    {{:@, meta, [{name, meta2, value}]}, represented}
+  end
+
+  # TODO: `when` clauses in specs but only for variable names, not key names in maps
+
```

Comment:
This seems a bit crazy already and it's not the full fix yet 🙊 I'm trying to fix the consequences of [`d2fc2d3` (#73)](https://github.com/exercism/elixir-representer/pull/73/commits/d2fc2d3df8e289b266c1c831729f4200df0a999b) but I'm not sure if it's worth it.

Reply:
Haha yeah it does seem a bit crazy and endless.
Representers get so complex after a while :(